### PR TITLE
Refactor of loading state access from SubViews

### PR DESCRIPTION
**Pull request by DavidKadlcek**
# :pencil: Description
- There is repeating adding parameters to Subviews when we need to get the loading state. Currently, there is a passing state via a constructor, but we can use it much easier via `EnvironmentObject`. So I create an `extension` of `EvnronmentValues` for this. You can just pass the `.environment` to the whole view and then you have access to it inside every subview.

# :bulb: What’s new?
- Custom EnvironmentValues - `isLoading`

### CODE REVIEW

**Review by tomas-bat — DISMISSED** [review]

File: ios/PresentationLayer/Onboarding/Sources/Onboarding/Login/LoginView.swift [hunk 1]
```diff
@@ -27,12 +27,12 @@ struct LoginView: View {
             Spacer()
             PrimaryAndSecondaryButtons(
                 primaryButtonTitle: L10n.login_view_login_button_title,
-                primaryButtonLoading: viewModel.state.loginButtonLoading,
                 onPrimaryButtonTap: { viewModel.onIntent(.login) },
                 secondaryButtonTitle: L10n.login_view_register_button_title,
                 onSecondaryButtonTap: { viewModel.onIntent(.register) }
             )
         }
+        .environment(\.isLoading, viewModel.state.loginButtonLoading)
```
Comment (pchmelar) [inline ios/PresentationLayer/Onboarding/Sources/Onboarding/Login/LoginView.swift]: Maybe change `viewModel.state.loginButtonLoading` to just `viewModel.state.loading`? As now it is used in context of the whole view, not just in context of the login button.

File: ios/PresentationLayer/Onboarding/Sources/Onboarding/Registration/RegistrationView.swift [hunk 2]
```diff
@@ -26,12 +26,12 @@ struct RegistrationView: View {
             Spacer()
             PrimaryAndSecondaryButtons(
                 primaryButtonTitle: L10n.login_view_register_button_title,
-                primaryButtonLoading: viewModel.state.registerButtonLoading,
                 onPrimaryButtonTap: { viewModel.onIntent(.register) },
                 secondaryButtonTitle: L10n.login_view_login_button_title,
                 onSecondaryButtonTap: { viewModel.onIntent(.login) }
             )
         }
+        .environment(\.isLoading, viewModel.state.registerButtonLoading)
```
Comment (pchmelar) [inline ios/PresentationLayer/Onboarding/Sources/Onboarding/Registration/RegistrationView.swift]: Same here

**Review by pchmelar — CHANGES_REQUESTED** [review]
Nice! 🚀

File: ios/PresentationLayer/Onboarding/Sources/Onboarding/Login/LoginView.swift (hunk 1 above)
Comment (DavidKadlcek) [reply ios/PresentationLayer/Onboarding/Sources/Onboarding/Login/LoginView.swift]: Done

File: ios/PresentationLayer/Onboarding/Sources/Onboarding/Registration/RegistrationView.swift (hunk 2 above)
Comment (DavidKadlcek) [reply ios/PresentationLayer/Onboarding/Sources/Onboarding/Registration/RegistrationView.swift]: Done

**Review by tomas-bat — DISMISSED** [review]

**Review by pchmelar — DISMISSED** [review]

**Review by pchmelar — APPROVED** [review]

**Pull request merged**